# 2. Finding $\alpha_{u,v}$

## 2.1 Generating Ulam numbers 

In [22]:
import Pkg; Pkg.add("Optim")
using Optim

   Resolving package versions...
  No Changes to `~/.julia/environments/v1.9/Project.toml`
  No Changes to `~/.julia/environments/v1.9/Manifest.toml`


In [3]:
function sums(L, k)
local S, i, j; 
S = []; 
for i in 1:length(L)-1
    for j in i+1:length(L) 
        if L[i] + L[j] == k
            S = union(S,[[L[i], L[j]]]); 
        end; end; end; return S; end;

function ulam(a_0, a_1, n) 
local u, k;
    if (n == 1) return [a_0]; 
    elseif (n == 2) return [a_0, a_1]; 
    else u = ulam(a_0, a_1, n - 1); 
    for k in u[n - 1] + 1: 1000000000000
        if (length(sums(u, k)) == 1)  return union(u, [k]); 
    end; end; end; end;

## 2.2 Computing the value $\alpha$ of $x$ for which $$f(x)=\sum_i(\cos( a_ix)) $$  is minimum.

In [4]:
function beta(u,v,n)
    Ulam=ulam(u,v,n)
    f(x)=sum(cos(x[1]*Ulam[i]) 
        for i=1:n)
    if u==1 & v>4
        xvalues=collect(range(2/v-0.2, 2/v+0.2, length=31500))#we change 0,pi to interval around 2/v.
    else 
        xvalues=collect(range(0,pi, length=31500))
    end
    gridsearch=findmin(f,xvalues)
    res=optimize(f,[xvalues[gridsearch[2]]])
    if Optim.minimum(res)<-0.6*n #this makes sure it's the minimum we're interested in.
    return Optim.minimizer(res)
    else print(bla) #if we get this we should refine the grid, but in practice, we never get this.
    end
end 

beta (generic function with 1 method)

## 2.3 How long does this take? 

In [42]:
L=[]
for n in 1:35
    print([n,@elapsed begin 
beta(2,3,100*n)
end],",")
end

[1.0, 0.050890266],[2.0, 0.11834318],[3.0, 0.227840882],[4.0, 0.347862787],[5.0, 0.578336777],[6.0, 0.843815187],[7.0, 1.113742127],[8.0, 1.521351197],[9.0, 2.05085871],[10.0, 2.672097046],[11.0, 3.400437168],[12.0, 4.29409003],[13.0, 5.37493734],[14.0, 7.267861195],[15.0, 7.997643094],[16.0, 9.59350568],[17.0, 11.579650133],[18.0, 14.388007227],[19.0, 16.17882418],[20.0, 19.13261061],[21.0, 21.2199024],[22.0, 24.243968881],[23.0, 27.458971985],[24.0, 32.012131231],[25.0, 36.284729702],[26.0, 38.40136662],[27.0, 43.055821853],[28.0, 51.938189073],[29.0, 54.790807028],[30.0, 59.606144933],[31.0, 64.897676784],[32.0, 79.697467594],[33.0, 86.264206636],[34.0, 91.790034284],[35.0, 106.138792051],

In [ ]:
![alternative text](link/to/image.png)

##  2.4 Approximating $\alpha_{u,v}$

### $\bullet$ When $u=1$, $\alpha_{1,v}\approx 2/v$

In [113]:
for v in 2:200
            print([v,beta(1,v,500)[1]],",")
end

[2.0, 2.571500155163835],[3.0, 2.8335073268512025],[4.0, 0.5059620791663552],[5.0, 0.4075223842841962],[6.0, 0.34119776716501105],[7.0, 0.2934240955859288],[8.0, 0.25731956716916937],[9.0, 0.2291939400599811],[10.0, 0.20655380759974798],[11.0, 0.18800286205964506],[12.0, 0.17254374077622597],[13.0, 0.1593785536187336],[14.0, 0.14810835552178936],[15.0, 0.1383342014200147],[16.0, 0.12975688251437573],[17.0, 0.12217692627218313],[18.0, 0.1154945964270923],[19.0, 0.1094106841800693],[20.0, 0.10402492579745878],[21.0, 0.09903811248022681],[22.0, 0.0946497167610627],[23.0, 0.09056052984093248],[24.0, 0.08677055171983619],[25.0, 0.08327978239777381],[26.0, 0.08008822187474535],[27.0, 0.07719587015075081],[28.0, 0.07440325469310091],[29.0, 0.07190984803448493],[30.0, 0.06951617764221359],[31.0, 0.06722224351628689],[32.0, 0.06518382680120659],[33.0, 0.06320771283383164],[34.0, 0.06133780380195317],[35.0, 0.05960255658901025],[36.0, 0.05794677074623543],[37.0, 0.056406841850759754],[38.0, 0.05

To compare against $\alpha_{2,2v}$.

In [15]:
for v in 2:200
            print([2*v,beta(1,v,500)[1]/2],",")
end

[4.0, 1.2857500775819175],[6.0, 1.4167536634256013],[8.0, 0.2529810395831776],[10.0, 0.2037611921420981],[12.0, 0.17059888358250552],[14.0, 0.1467120477929644],[16.0, 0.12865978358458469],[18.0, 0.11459697002999054],[20.0, 0.10327690379987399],[22.0, 0.09400143102982253],[24.0, 0.08627187038811299],[26.0, 0.0796892768093668],[28.0, 0.07405417776089468],[30.0, 0.06916710071000735],[32.0, 0.06487844125718786],[34.0, 0.061088463136091564],[36.0, 0.05774729821354615],[38.0, 0.05470534209003465],[40.0, 0.05201246289872939],[42.0, 0.04951905624011341],[44.0, 0.04732485838053135],[46.0, 0.04528026492046624],[48.0, 0.043385275859918096],[50.0, 0.041639891198886905],[52.0, 0.04004411093737267],[54.0, 0.03859793507537541],[56.0, 0.037201627346550456],[58.0, 0.035954924017242464],[60.0, 0.03475808882110679],[62.0, 0.033611121758143445],[64.0, 0.032591913400603294],[66.0, 0.03160385641691582],[68.0, 0.030668901900976583],[70.0, 0.029801278294505126],[72.0, 0.028973385373117715],[74.0, 0.0282034209

LoadError: InterruptException:

### $\bullet$ When $u$ or $v$ is even, $\alpha_{u,v}\approx \pi$ 

In [112]:
for v in 3:100
            print([v,beta(2,v,500)[1]],",")
end

[3.0, 1.1650193271717315],[4.0, 1.855892444141048],[5.0, 3.141592653589793],[6.0, 1.7248389901641918],[7.0, 3.141592653589793],[8.0, 0.2530309077163499],[9.0, 3.141592653589793],[10.0, 0.2037611921420981],[11.0, 3.141592653589793],[12.0, 2.97104363814046],[13.0, 3.141592653589793],[14.0, 2.994880605796829],[15.0, 3.141592653589793],[16.0, 0.12865978358458469],[17.0, 3.141592653589793],[18.0, 0.11459697002999054],[19.0, 3.141592653589793],[20.0, 0.1033267719330463],[21.0, 3.141592653589793],[22.0, 0.09405129916299486],[23.0, 3.141592653589793],[24.0, 0.08627187038811299],[25.0, 3.141592653589793],[26.0, 0.0796892768093668],[27.0, 3.141592653589793],[28.0, 0.074104045894067],[29.0, 3.141592653589793],[30.0, 0.06921696884317967],[31.0, 3.141592653589793],[32.0, 0.06492830939036018],[33.0, 3.141592653589793],[34.0, 3.0804543223205294],[35.0, 3.141592653589793],[36.0, 0.05774729821354615],[37.0, 3.141592653589793],[38.0, 0.05475521022320697],[39.0, 3.141592653589793],[40.0, 0.05196259476555

If $v$ is even then the $Ulam(2,2v)$ is just $2\cdot Ulam(1,v,2)$, so we plot the values of $\alpha_{2,2v}$ with the values of $\alpha_{1,v}$.

In [28]:
for v in 2:100
            print([2*v,beta(2,2*v,500)[1]],",")
end

[4.0, 1.855892444141048],
[6.0, 1.7248389901641918],
[8.0, 0.2530309077163499],
[10.0, 0.2037611921420981],
[12.0, 2.97104363814046],
[14.0, 2.994880605796829],
[16.0, 0.12865978358458469],
[18.0, 0.11459697002999054],
[20.0, 0.1033267719330463],
[22.0, 0.09405129916299486],
[24.0, 0.08627187038811299],
[26.0, 0.0796892768093668],
[28.0, 0.074104045894067],
[30.0, 0.06921696884317967],
[32.0, 0.06492830939036018],
[34.0, 3.0804543223205294],
[36.0, 0.05774729821354615],
[38.0, 0.05475521022320697],
[40.0, 0.05196259476555707],
[42.0, 0.04956892437328573],
[44.0, 3.094317663342434],
[46.0, 0.04528026492046624],
[48.0, 0.043385275859918096],
[50.0, 0.04168975933205923],
[52.0, 3.1014986745192483],
[54.0, 0.03859793507537541],
[56.0, 0.037201627346550456],
[58.0, 3.105687597705723],
[60.0, 0.03470822068793447],
[62.0, 3.1079815318316495],
[64.0, 3.1089788944950962],
[66.0, 0.03161639643125066],
[68.0, 3.1108738835556444],
[70.0, 0.02982114363704715],
[72.0, 0.029023253506290033],
[74.0, 3

When $u=2$ we notice that for $v$ odd we get $\alpha_{2,v}=\pi$

In [13]:
for v in 1:50
            println([2,2*v+1],beta(2,2*v+1,500))
end

[2, 3][1.1650193271717315]
[2, 5][3.141592653589793]
[2, 7][3.141592653589793]
[2, 9][3.141592653589793]
[2, 11][3.141592653589793]
[2, 13][3.141592653589793]
[2, 15][3.141592653589793]
[2, 17][3.141592653589793]
[2, 19][3.141592653589793]
[2, 21][3.141592653589793]
[2, 23][3.141592653589793]
[2, 25][3.141592653589793]
[2, 27][3.141592653589793]
[2, 29][3.141592653589793]
[2, 31][3.141592653589793]
[2, 33][3.141592653589793]
[2, 35][3.141592653589793]
[2, 37][3.141592653589793]
[2, 39][3.141592653589793]
[2, 41][3.141592653589793]
[2, 43][3.141592653589793]
[2, 45][3.141592653589793]
[2, 47][3.141592653589793]
[2, 49][3.141592653589793]
[2, 51][3.141592653589793]
[2, 53][3.141592653589793]
[2, 55][3.141592653589793]
[2, 57][3.141592653589793]
[2, 59][3.141592653589793]
[2, 61][3.141592653589793]
[2, 63][3.141592653589793]
[2, 65][3.141592653589793]
[2, 67][3.141592653589793]
[2, 69][3.141592653589793]
[2, 71][3.141592653589793]
[2, 73][3.141592653589793]
[2, 75][3.141592653589793]
[2, 

In [14]:
for v in 1:50
            println([4,2*v+1],beta(4,2*v+1,500))
end

[4, 3][2.2090585632674156]
[4, 5][3.141592653589793]
[4, 7][3.141592653589793]
[4, 9][3.141592653589793]
[4, 11][3.141592653589793]
[4, 13][3.141592653589793]
[4, 15][3.141592653589793]
[4, 17][3.141592653589793]
[4, 19][3.141592653589793]
[4, 21][3.141592653589793]
[4, 23][3.141592653589793]
[4, 25][3.141592653589793]
[4, 27][3.141592653589793]
[4, 29][3.141592653589793]
[4, 31][3.141592653589793]
[4, 33][3.141592653589793]
[4, 35][3.141592653589793]
[4, 37][3.141592653589793]
[4, 39][3.141592653589793]
[4, 41][3.141592653589793]
[4, 43][3.141592653589793]
[4, 45][3.141592653589793]
[4, 47][3.141592653589793]
[4, 49][3.141592653589793]
[4, 51][3.141592653589793]
[4, 53][3.141592653589793]
[4, 55][3.141592653589793]
[4, 57][3.141592653589793]
[4, 59][3.141592653589793]
[4, 61][3.141592653589793]
[4, 63][3.141592653589793]
[4, 65][3.141592653589793]
[4, 67][3.141592653589793]
[4, 69][3.141592653589793]
[4, 71][3.141592653589793]
[4, 73][3.141592653589793]
[4, 75][3.141592653589793]
[4, 

We check whether this holds more generally for $u$ even.

Indeed, for $u$ even and $v$ coprime to $u$ we have that $\alpha\approx \pi$. The following command tests this.

In [29]:
for u in 1:40
for v in 2*u+1:2*u+40
        if gcd(2*u,v)==1
            println([2*u,v],beta(2*u,v,500))
        end
end
end

[2, 3][1.1650193271717315]
[2, 5][3.141592653589793]
[2, 7][3.141592653589793]
[2, 9][3.141592653589793]
[2, 11][3.141592653589793]
[2, 13][3.141592653589793]
[2, 15][3.141592653589793]
[2, 17][3.141592653589793]
[2, 19][3.141592653589793]
[2, 21][3.141592653589793]
[2, 23][3.141592653589793]
[2, 25][3.141592653589793]
[2, 27][3.141592653589793]
[2, 29][3.141592653589793]
[2, 31][3.141592653589793]
[2, 33][3.141592653589793]
[2, 35][3.141592653589793]
[2, 37][3.141592653589793]
[2, 39][3.141592653589793]
[2, 41][3.141592653589793]
[4, 5][3.141592653589793]
[4, 7][3.141592653589793]
[4, 9][3.141592653589793]
[4, 11][3.141592653589793]
[4, 13][3.141592653589793]
[4, 15][3.141592653589793]
[4, 17][3.141592653589793]
[4, 19][3.141592653589793]
[4, 21][3.141592653589793]
[4, 23][3.141592653589793]
[4, 25][3.141592653589793]
[4, 27][3.141592653589793]
[4, 29][3.141592653589793]
[4, 31][3.141592653589793]
[4, 33][3.141592653589793]
[4, 35][3.141592653589793]
[4, 37][3.141592653589793]
[4, 39]

For instance, the minimum of $f$ when $(u,v)=(4,7)$ is $-492$, which tells us that there are exactly $4$ even numbers out of $500$ in the Ulam sequence (this is done solving $x+y=500,-x+y=-492$).

In [37]:
beta2(4,7,500)

-492.0

The equivalent statement fails when $v$ is even and $v$ coprime to $u$ if $u<7$ but holds when $u>5$.

In [86]:
for u in 1:20
for v in u+1:u+20 
        if gcd(2*u+1,2*v)==1
            println([2*u+1,2*v],beta(2*u+1,2*v,500))
        end
end
end

[3, 4][2.2090585632674156]
[3, 8][2.0338219432998845]
[3, 10][2.143731308811677]
[3, 14][2.058556537353355]
[3, 16][2.1258785171359866]
[3, 20][2.0689291090531974]
[3, 22][2.1176004070293817]
[3, 26][2.074614076234842]
[3, 28][2.112813066244839]
[3, 32][2.0783043180895935]
[3, 34][2.1096215057218104]
[3, 38][2.0807977247482095]
[3, 40][2.1073275715958837]
[5, 6][3.141592653589793]
[5, 8][1.228750801365956]
[5, 12][1.2762252641460043]
[5, 14][2.496498482872666]
[5, 16][2.5281148793039168]
[5, 18][1.2432125599859287]
[5, 22][1.2676479452403655]
[5, 24][2.503081076451412]
[5, 26][2.5227291209213063]
[5, 28][1.2479001645041268]
[5, 32][1.2643566484509923]
[5, 34][2.5059734281754067]
[5, 36][2.5201359779963455]
[5, 38][1.2500943623637089]
[5, 42][1.2626611319231336]
[5, 44][2.507569208436921]
[7, 8][3.141592653589793]
[7, 10][3.141592653589793]
[7, 12][3.141592653589793]
[7, 16][3.141592653589793]
[7, 18][3.141592653589793]
[7, 20][3.141592653589793]
[7, 22][3.141592653589793]
[7, 24][3.141

Asking for $u,v$ to be even and odd respectivly is not enough, e.g. for $(6,9)$, $f$ does not minimize at $\pi$:

In [88]:
for u in 1:20 #(even,odd)
for v in u:u+10
        #if gcd(2*u,2*v+1)==1
            println([2*u,2*v+1], beta(2*u,2*v+1,500))
        #end
end
end

[2, 3][1.1650193271717315]
[2, 5][3.141592653589793]
[2, 7][3.141592653589793]
[2, 9][3.141592653589793]
[2, 11][3.141592653589793]
[2, 13][3.141592653589793]
[2, 15][3.141592653589793]
[2, 17][3.141592653589793]
[2, 19][3.141592653589793]
[2, 21][3.141592653589793]
[2, 23][3.141592653589793]
[4, 5][3.141592653589793]
[4, 7][3.141592653589793]
[4, 9][3.141592653589793]
[4, 11][3.141592653589793]
[4, 13][3.141592653589793]
[4, 15][3.141592653589793]
[4, 17][3.141592653589793]
[4, 19][3.141592653589793]
[4, 21][3.141592653589793]
[4, 23][3.141592653589793]
[4, 25][3.141592653589793]
[6, 7][3.141592653589793]
[6, 9][2.482734878117106]
[6, 11][3.141592653589793]
[6, 13][3.141592653589793]
[6, 15][3.141592653589793]
[6, 17][3.141592653589793]
[6, 19][3.141592653589793]
[6, 21][3.141592653589793]
[6, 23][3.141592653589793]
[6, 25][3.141592653589793]
[6, 27][3.141592653589793]
[8, 9][3.141592653589793]
[8, 11][3.141592653589793]
[8, 13][3.141592653589793]
[8, 15][3.141592653589793]
[8, 17][3.

LoadError: UndefVarError: `bla` not defined

###  $\bullet$ When $u=3$, $$\lim_{v\rightarrow \infty} \alpha_{3,v}\approx 2.1$$ 

In [33]:
for v in 150:250
    if gcd(3,v)==1
            println([v,beta(3,v,1000)[1]],",")
end
end

[151.0, 2.0982515713585213],
[152.0, 2.0909708239153626],
[154.0, 2.0978526262931427],
[155.0, 2.090671615116329],
[157.0, 2.09812620411961],
[158.0, 2.0910705601817074],
[160.0, 2.0976531537604535],
[161.0, 2.0907713513826733],
[163.0, 2.098052098825832],
[164.0, 2.091170296448052],
[166.0, 2.097553417494109],
[167.0, 2.090871087649018],
[169.0, 2.0978526262931427],
[170.0, 2.0912700327143967],
[172.0, 2.097453681227764],
[173.0, 2.0909708239153626],
[175.0, 2.0977528900267983],
[176.0, 2.0913697689807416],
[178.0, 2.09735394496142],
[179.0, 2.0910705601817074],
[181.0, 2.0976531537604535],
[182.0, 2.091469505247086],
[184.0, 2.097254208695075],
[185.0, 2.091170296448052],
[187.0, 2.097553417494109],
[188.0, 2.091569241513431],
[190.0, 2.0971544724287305],
[191.0, 2.0912700327143967],
[193.0, 2.097453681227764],
[194.0, 2.0916689777797752],
[196.0, 2.0971544724287305],
[197.0, 2.0913697689807416],
[199.0, 2.09735394496142],
[200.0, 2.09176871404612],
[202.0, 2.0970547361623857],
[203.

In [35]:
for v in 150:250
    if gcd(3,v)==1
            print([v,beta(3,v,500)[1]],",")
end
end

[151.0, 2.0982515713585213],[152.0, 2.090871087649018],[154.0, 2.0978526262931427],[155.0, 2.090671615116329],[157.0, 2.098151835092177],[158.0, 2.0910705601817074],[160.0, 2.0977528900267983],[161.0, 2.0907713513826733],[163.0, 2.0979523625594876],[164.0, 2.091170296448052],[166.0, 2.0976531537604535],[167.0, 2.090871087649018],[169.0, 2.0978526262931427],[170.0, 2.091170296448052],[172.0, 2.097553417494109],[173.0, 2.0909708239153626],[175.0, 2.0977528900267983],[176.0, 2.0912700327143967],[178.0, 2.097453681227764],[179.0, 2.0910705601817074],[181.0, 2.0976531537604535],[182.0, 2.0913697689807416],[184.0, 2.09735394496142],[185.0, 2.091170296448052],[187.0, 2.097553417494109],[188.0, 2.091469505247086],[190.0, 2.097254208695075],[191.0, 2.0912700327143967],[193.0, 2.097453681227764],[194.0, 2.091569241513431],[196.0, 2.0971544724287305],[197.0, 2.0913697689807416],[199.0, 2.097453681227764],[200.0, 2.0916689777797752],[202.0, 2.0971544724287305],[203.0, 2.091469505247086],[205.0, 2.

# 3. Equidistribution

#### We study the equidsitribution of $ (x_i)_{i=1,...,N}$ where $$x_i=\alpha_{u,v} \cdot Ulam(u,v,500)[i] / 2\pi$$  

#### We expect $$\langle x_i\rangle\in [1/4,3/4]$$ for most $i$.


In [6]:
function rangle_list(u,v,n)
    L = []
    ulamseq=ulam(u,v,n)
    alpha=beta(u,v,n)[1]
    for i in 1:n
        ranglevalue = 0.5 * alpha * ulamseq[i] / pi - floor(0.5 * alpha * ulamseq[i] / pi)
        push!(L, ranglevalue)
    end
    return L
end

rangle_list (generic function with 1 method)

The following function lists and counts for each pair of coprime elements $u,v$ the elements outside of the interval $I=[1/4,3/4]$.

In [38]:
function ran()
H=[]
n=0
for u in 1:20
    for v in u+1:u+20 
        if gcd(u,v)==1 #imposing u,v coprime 
            n=n+1 #counts number of coprime pairs u,v
            r=rangle_list(u,v,500)
            l=0 #counts number of elements outside interval [0.25,0.75]
            N=[] #This is the list of elements outside interval, concretely, it lists the position of the outcast inside $[x_n]_n$ and its value.
            
            for i in 1:500
                if r[i] >= 0.75 
                    l=l+1
                    N=union(N,[[i,r[i]]])
                elseif r[i] <= 0.25
                    l=l+1 
                    N=union(N,[[i,r[i]]])  
                end
            end
                if l>4 
                    H=union(H,[[[u,v],l]]) #this is the list of the $(u,v)$ for which there are more than 4 elements outside of $I$.
                end
            println([[u,v],l,N])
            println()
        end
    end
end
    print(["number of pairs for which there are more than 4 let's outside I=",length(H),H,"number of coprime pairs=",n])
end

ran (generic function with 1 method)

In [9]:
ran()

Any[[1, 2], 4, Any[[2.0, 0.8185226068044459], [3.0, 0.22778391020666877], [15.0, 0.2352812599044789], [20.0, 0.23902993475338263]]]

Any[[1, 3], 2, Any[[3.0, 0.8038667089257805], [27.0, 0.19473436868162963]]]

Any[[1, 4], 2, Any[[1.0, 0.08053639930491042], [7.0, 0.8053639930491043]]]

Any[[1, 5], 2, Any[[1.0, 0.06486409687537568], [8.0, 0.7783691625045083]]]

Any[[1, 6], 2, Any[[1.0, 0.054295675362572265], [9.0, 0.7601394550760118]]]

Any[[1, 7], 1, Any[[1.0, 0.04669449939343599]]]

Any[[1, 8], 1, Any[[1.0, 0.040956731767560296]]]

Any[[1, 9], 1, Any[[1.0, 0.036470343938185115]]]

Any[[1, 10], 1, Any[[1.0, 0.03287467622596968]]]

Any[[1, 11], 1, Any[[1.0, 0.029923824304090754]]]

Any[[1, 12], 1, Any[[1.0, 0.027459316495262245]]]

Any[[1, 13], 1, Any[[1.0, 0.025371861987562708]]]

Any[[1, 14], 1, Any[[1.0, 0.02357402813145499]]]

Any[[1, 15], 1, Any[[1.0, 0.022016634061574447]]]

Any[[1, 16], 1, Any[[1.0, 0.02065596345315249]]]

Any[[1, 17], 1, Any[[1.0, 0.01944829998142056]]]

Any[[1, 

LoadError: InterruptException:

Each list in the output corresponds to $[[u,v],$ number of outcasts, list of outcasts$=[[n,x_n]:\langle x_n\rangle\notin I]]$.

Very often the first element $\langle x_1\rangle$ in $[\langle x_i\rangle]_i$ is $0$ and, in particular, outside of $I$. 

To investigate this we make a function which returns $\langle x_1\rangle$ for given $u,v$.

In [31]:
function firstel(u,v)
    alpha=beta(u,v,500)[1]
    ranglevalue = 0.5 * alpha * u / pi - floor(0.5 * alpha * u / pi)
    return ranglevalue
end

firstel (generic function with 1 method)

## 3.1 For which $(u,v)$ is $\langle x_1\rangle$ not an outcast?

###### 

In [11]:
n=0
L=[]
for u in 1:20
    for v in u+1:u+20 
        if gcd(u,v)==1
            f=firstel(u,v)
            println([u,v],f)
            if (f>=0.25) & (f<=0.75)
                n=n+1 #counting for how many pairs (u,v) we have that <x_1> is in I
                L=union(L,[[[u,v],f,beta(u,v,500)]])
            end
        end
    end
end
print(n,L)

[1, 2]0.4092669608559002
[1, 3]0.4509666973554716
[1, 4]0.0805263659163783
[1, 5]0.06485920187942475
[1, 6]0.05430331121622909
[1, 7]0.046699895234769355
[1, 8]0.04095368106924029
[1, 9]0.036477348487253564
[1, 10]0.03287405949395219
[1, 11]0.02992158481221626
[1, 12]0.027461189244102988
[1, 13]0.025365884631258138
[1, 14]0.02357217689450459
[1, 15]0.022016571954665227
[1, 16]0.020651449252357217
[1, 17]0.019445061747991998
[1, 18]0.018381535921775292
[1, 19]0.017413251214324264
[1, 20]0.01655608114543319
[1, 21]0.01576240515571923
[2, 3]0.37083716943395023
[2, 5]0.0
[2, 7]0.0
[2, 9]0.0
[2, 11]0.0
[2, 13]0.0
[2, 15]0.0
[2, 17]0.0
[2, 19]0.0
[2, 21]0.0
[3, 4]0.0547477697704688
[3, 5]0.9572684847138004
[3, 7]0.033889964760785984
[3, 8]0.9710784469348234
[3, 10]0.023556303374710152
[3, 11]0.9784120130797803
[3, 13]0.018889488555192235
[3, 14]0.982888345661767
[3, 16]0.015032223245182497
[3, 17]0.9851741325121431
[3, 19]0.01331788310740012
[3, 20]0.9878408838375821
[3, 22]0.011079716816406

#### As expected, for $u>5$ odd and $v$ even we have $\alpha\approx\pi$ so $$\langle x_1\rangle\approx\langle \pi\cdot u /2\pi\rangle=\langle u \rangle =1/2$$ is not an outcast

Other than that, we only get a few other pairs $[u,v]$:
$$[1, 2],$$
$$ [1, 3],$$
$$   [2, 3],$$
$$   [5, 6],$$
$$    [5, 7],$$
$$[9, 11],$$
   $$ [9, 13],$$
   $$[57, 59]$$

##  3.2 A pattern in the lists of outcasts when $\langle x_1\rangle\notin I$

#### 

In [32]:
function ran4()
H=[]
n=0
for u in 2:20
    for v in u+1:2*u+20 
        f=firstel(u,v)
        if (gcd(u,v)==1) & xor(abs(f)<0.1,abs(1-f)<0.1) #imposing u,v coprime and that [1,0] is in the list given by ran()
            n=n+1 #counts number of coprime pairs u,v
            r=rangle_list(u,v,500)
            l=0 #counts number of elements outisde interval [0.25,0.75]
            N=[] #lists elements outside interval
            for i in 2:500
        
                if xor(r[i] >= 0.75 ,r[i] <= 0.25) 
                    #l=l+1
                    N=union(N,[i])
                    #return [[u,v],i,r[i]]
                    #println([[u,v],i,r[i]])
                break
                    
                    end
                    
            end
            
               # if l>4 
               #     H=union(H,[[[u,v],l,n]]) 
               # end
            println( [[u,v],N])
               # println([[u,v],i,r[i]])
           # println()
        end
    end
end
   # print([length(H),H])
end

ran4 (generic function with 1 method)

In [33]:
ran4()

Vector{Any}[[2, 5], [6]]
Vector{Any}[[2, 7], [7]]
Vector{Any}[[2, 9], [8]]
Vector{Any}[[2, 11], [9]]
Vector{Any}[[2, 13], [10]]
Vector{Any}[[2, 15], [11]]
Vector{Any}[[2, 17], [12]]
Vector{Any}[[2, 19], [13]]
Vector{Any}[[2, 21], [14]]
Vector{Any}[[2, 23], [15]]
Vector{Any}[[3, 4], [5]]
Vector{Any}[[3, 5], [5]]
Vector{Any}[[3, 7], [6]]
Vector{Any}[[3, 8], [6]]
Vector{Any}[[3, 10], [7]]
Vector{Any}[[3, 11], [7]]
Vector{Any}[[3, 13], [8]]
Vector{Any}[[3, 14], [8]]
Vector{Any}[[3, 16], [9]]
Vector{Any}[[3, 17], [9]]
Vector{Any}[[3, 19], [10]]
Vector{Any}[[3, 20], [10]]
Vector{Any}[[3, 22], [11]]
Vector{Any}[[3, 23], [11]]
Vector{Any}[[3, 25], [12]]
Vector{Any}[[3, 26], [12]]
Vector{Any}[[4, 5], [5]]
Vector{Any}[[4, 7], [5]]
Vector{Any}[[4, 9], [6]]
Vector{Any}[[4, 11], [6]]
Vector{Any}[[4, 13], [7]]
Vector{Any}[[4, 15], [7]]
Vector{Any}[[4, 17], [8]]
Vector{Any}[[4, 19], [8]]
Vector{Any}[[4, 21], [9]]
Vector{Any}[[4, 23], [9]]
Vector{Any}[[4, 25], [10]]
Vector{Any}[[4, 27], [10]]
Vector{A

LoadError: UndefVarError: `bla` not defined

#### When $u=3$


when $v=4$, $(\langle x_i \rangle)_i=[0.055, 0.406, 0.461, 0.516, \underline{\textbf{0.867}}, 0.571, 0.274, 0.625]$

when $v=5$, $(\langle x_i \rangle)_i=[0.957, 0.595, 0.553, 0.51, \underline{\textbf{0.148}}, 0.467, 0.424, 0.744]$

when $v=7$, $(\langle x_i \rangle)_i=[0.034, 0.412, 0.446, 0.48, 0.514, \underline{\textbf{0.859}}, 0.548, 0.582]$

when $v=8$, $(\langle x_i \rangle)_i=[0.971, 0.589, 0.561, 0.532, 0.503, \underline{\textbf{0.15}}, 0.474, 0.445]$

when $v=10$, $(\langle x_i \rangle)_i=[0.024, 0.412, 0.435, 0.459, 0.483, 0.506, \underline{\textbf{0.847}}, 0.53]$

when $v=11$, $(\langle x_i \rangle)_i=[0.978, 0.587, 0.566, 0.544, 0.523, 0.501, \underline{\textbf{0.153}}, 0.479]$




In the original sequence...



$Ulam(3,4)=[3, 4, 7, 10, \underline{\textbf{11}}, 13, 15, 16]$

$Ulam(3,5)=[3, 5, 8, 11, \underline{\textbf{13}}, 14, 17, 18]$

$Ulam(3,7)=[3, 7, 10, 13, 16, \underline{\textbf{17}}, 19, 22]$

$Ulam(3,8)=[3, 8, 11, 14, 17, \underline{\textbf{19}}, 20, 23]$

$Ulam(3,10)=[3, 10, 13, 16, 19, 22, \underline{\textbf{23}}, 25]$

$Ulam(3,11)=[3, 11, 14, 17, 20, 23, \underline{\textbf{25}}, 26]$

In [128]:
for v in 4:100 
        f=firstel(3,v)
        r=rangle_list(3,v,500)
        
        if (gcd(3,v)==1) & xor(abs(f)<0.1,abs(1-f)<0.1)
        N=[]
        for i in 2:500
        
                if xor(r[i] >= 0.75 ,r[i] <= 0.25) 
                    #l=l+1
                    N=union(N,[i])
                    #return [[u,v],i,r[i]]
                    #println([[u,v],i,r[i]])
                break
                    
                end
                    
        end
        println( [[3,v],N])
    end
            
               # if l>4 
               #     H=union(H,[[[u,v],l,n]]) 
               # end
            
end

Vector{Any}[[3, 4], [5]]
Vector{Any}[[3, 5], [5]]
Vector{Any}[[3, 7], [6]]
Vector{Any}[[3, 8], [6]]
Vector{Any}[[3, 10], [7]]
Vector{Any}[[3, 11], [7]]
Vector{Any}[[3, 13], [8]]
Vector{Any}[[3, 14], [8]]
Vector{Any}[[3, 16], [9]]
Vector{Any}[[3, 17], [9]]
Vector{Any}[[3, 19], [10]]
Vector{Any}[[3, 20], [10]]
Vector{Any}[[3, 22], [11]]
Vector{Any}[[3, 23], [11]]
Vector{Any}[[3, 25], [12]]
Vector{Any}[[3, 26], [12]]
Vector{Any}[[3, 28], [13]]
Vector{Any}[[3, 29], [13]]
Vector{Any}[[3, 31], [14]]
Vector{Any}[[3, 32], [14]]
Vector{Any}[[3, 34], [15]]
Vector{Any}[[3, 35], [15]]
Vector{Any}[[3, 37], [16]]
Vector{Any}[[3, 38], [16]]
Vector{Any}[[3, 40], [17]]
Vector{Any}[[3, 41], [17]]
Vector{Any}[[3, 43], [18]]
Vector{Any}[[3, 44], [18]]
Vector{Any}[[3, 46], [19]]
Vector{Any}[[3, 47], [19]]
Vector{Any}[[3, 49], [20]]
Vector{Any}[[3, 50], [20]]
Vector{Any}[[3, 52], [21]]
Vector{Any}[[3, 53], [21]]
Vector{Any}[[3, 55], [22]]
Vector{Any}[[3, 56], [22]]
Vector{Any}[[3, 58], [23]]
Vector{Any}[[3,

## 3.3 Comparing Ulam(3,v) to the boring sequence 

#### 

We first generate the sequence $$(n,n+v,n+n+v,...)$$

In [2]:
function list(n,v)
L=[n,v]
for k in 1:498
    L=push!(L,n*k+v)
end
return L
end

list (generic function with 1 method)

#### Matrix out of $Ulam(3,v)$

In [167]:
for v in 4:50
if gcd(3,v)==1
    S=ulam(3,v,500)
    println(S[1:30],",")
    
    end  
end

[3, 4, 7, 10, 11, 13, 15, 16, 21, 22, 27, 30, 35, 36, 41, 44, 50, 53, 55, 61, 69, 70, 75, 78, 84, 87, 92, 93, 98, 101],
[3, 5, 8, 11, 13, 14, 17, 18, 20, 24, 26, 30, 39, 40, 45, 46, 49, 67, 71, 73, 74, 77, 92, 98, 99, 102, 114, 124, 134, 135],
[3, 7, 10, 13, 16, 17, 19, 22, 24, 25, 28, 30, 34, 36, 42, 51, 56, 60, 62, 65, 77, 88, 89, 91, 106, 109, 114, 115, 126, 129],
[3, 8, 11, 14, 17, 19, 20, 23, 26, 27, 29, 32, 33, 38, 39, 45, 51, 54, 60, 63, 67, 73, 75, 76, 82, 85, 88, 91, 103, 110],
[3, 10, 13, 16, 19, 22, 23, 25, 28, 31, 33, 34, 37, 39, 40, 45, 46, 51, 57, 63, 66, 72, 75, 83, 89, 92, 93, 98, 101, 104],
[3, 11, 14, 17, 20, 23, 25, 26, 29, 32, 35, 36, 38, 41, 42, 44, 48, 50, 54, 60, 66, 75, 84, 87, 88, 94, 97, 99, 109, 133],
[3, 13, 16, 19, 22, 25, 28, 29, 31, 34, 37, 40, 42, 43, 46, 48, 49, 52, 54, 58, 60, 66, 72, 78, 87, 96, 99, 104, 110, 113],
[3, 14, 17, 20, 23, 26, 29, 31, 32, 35, 38, 41, 44, 45, 47, 50, 51, 53, 56, 57, 62, 63, 69, 75, 81, 87, 90, 96, 99, 115],
[3, 16, 19, 22, 

#### Matrix out of $Ulam(3,v)-(3,3+v,3+3+v, 3+3+3+v, ...)$ for each $v$ coprime to $3$.

In [163]:
for v in 4:50
    if gcd(3,v)==1
        
        S=ulam(3,v,500)-list(3,v)
        println(S[1:30],",")
    end
end

[3, 3, 3, 3, 1, 0, -1, -3, -1, -3, -1, -1, 1, -1, 1, 1, 4, 4, 3, 6, 11, 9, 11, 11, 14, 14, 16, 14, 16, 16],
[3, 3, 3, 3, 2, 0, 0, -2, -3, -2, -3, -2, 4, 2, 4, 2, 2, 17, 18, 17, 15, 15, 27, 30, 28, 28, 37, 44, 51, 49],
[3, 3, 3, 3, 3, 1, 0, 0, -1, -3, -3, -4, -3, -4, -1, 5, 7, 8, 7, 7, 16, 24, 22, 21, 33, 33, 35, 33, 41, 41],
[3, 3, 3, 3, 3, 2, 0, 0, 0, -2, -3, -3, -5, -3, -5, -2, 1, 1, 4, 4, 5, 8, 7, 5, 8, 8, 8, 8, 17, 21],
[3, 3, 3, 3, 3, 3, 1, 0, 0, 0, -1, -3, -3, -4, -6, -4, -6, -4, -1, 2, 2, 5, 5, 10, 13, 13, 11, 13, 13, 13],
[3, 3, 3, 3, 3, 3, 2, 0, 0, 0, 0, -2, -3, -3, -5, -6, -5, -6, -5, -2, 1, 7, 13, 13, 11, 14, 14, 13, 20, 41],
[3, 3, 3, 3, 3, 3, 3, 1, 0, 0, 0, 0, -1, -3, -3, -4, -6, -6, -7, -6, -7, -4, -1, 2, 8, 14, 14, 16, 19, 19],
[3, 3, 3, 3, 3, 3, 3, 2, 0, 0, 0, 0, 0, -2, -3, -3, -5, -6, -6, -8, -6, -8, -5, -2, 1, 4, 4, 7, 7, 20],
[3, 3, 3, 3, 3, 3, 3, 3, 1, 0, 0, 0, 0, 0, -1, -3, -3, -4, -6, -6, -7, -9, -7, -9, -7, -4, -1, 2, 5, 5],
[3, 3, 3, 3, 3, 3, 3, 3, 2, 0, 0, 0, 0

#### Matrix out of $(\langle x_n\rangle)_n$

In [46]:
for v in 4:50
    if gcd(3, v) == 1
        S = rangle_list(3, v, 500)[1:30]
        println(S,",")
    end
end

Any[0.0547477697704688, 0.4063303596939585, 0.4610781294644273, 0.5158258992348963, 0.8674084891583855, 0.5705736690053653, 0.2737388488523438, 0.6253214387758339, 0.3832343883932827, 0.7348169783167711, 0.492729927934219, 0.5474776977046876, 0.3053906473221364, 0.6569732372456265, 0.4148861868630753, 0.4696339566335421, 0.5791294961744811, 0.6338772659449496, 0.3370424457919263, 0.44653798533286704, 0.25919870472078443, 0.6107812946442728, 0.3686942442617216, 0.4234420140321902, 0.5329375535731309, 0.5876853233435995, 0.3455982729610412, 0.6971808628845366, 0.45509381250198544, 0.5098415822724505],
Any[0.9572684847138004, 0.5954474745230007, 0.5527159592368012, 0.5099844439506018, 0.14816343375980168, 0.4672529286644025, 0.42452141337820226, 0.7436109082828022, 0.3817898980920029, 0.6581478777104035, 0.29632686751960335, 0.5726848471380048, 0.44449030127940503, 0.7635797961840058, 0.35902727070700813, 0.6781167656116072, 0.6353852503254078, 0.37899615860821356, 0.6553541382266133, 0.2

We do the same but for general $v$.

In [169]:
for v in 4:50
        
        S=ulam(3,2*v+1,500)-list(3,v)
        println(S[1:30],",")
        
    
end

[3, 8, 8, 8, 8, 11, 14, 17, 29, 38, 41, 53, 62, 65, 77, 86, 98, 107, 110, 122, 131, 143, 152, 155, 167, 176, 218, 218, 221, 245],
[3, 9, 9, 9, 9, 9, 8, 6, 6, 6, 6, 4, 3, 3, 1, 0, 1, 0, 1, 4, 7, 13, 19, 19, 17, 20, 20, 19, 26, 47],
[3, 10, 10, 10, 10, 10, 10, 8, 7, 7, 7, 7, 6, 4, 4, 3, 1, 1, 0, 1, 0, 3, 6, 9, 15, 21, 21, 23, 26, 26],
[3, 11, 11, 11, 11, 11, 11, 14, 35, 38, 38, 38, 41, 74, 74, 74, 74, 104, 116, 149, 149, 149, 179, 191, 221, 230, 260, 275, 302, 302],
[3, 12, 12, 12, 12, 12, 12, 12, 11, 9, 9, 9, 9, 9, 9, 7, 6, 6, 4, 3, 3, 1, 0, 1, 0, 1, 4, 7, 10, 13],
[3, 13, 13, 13, 13, 13, 13, 13, 13, 11, 10, 10, 10, 10, 10, 10, 9, 7, 7, 6, 4, 4, 3, 1, 1, 0, 1, 0, 3, 6],
[3, 14, 14, 14, 14, 14, 14, 14, 14, 17, 50, 53, 53, 53, 53, 53, 56, 101, 101, 101, 101, 101, 101, 143, 161, 206, 206, 206, 206, 206],
[3, 15, 15, 15, 15, 15, 15, 15, 15, 15, 14, 12, 12, 12, 12, 12, 12, 12, 12, 10, 9, 9, 7, 6, 6, 4, 3, 3, 1, 0],
[3, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 14, 13, 13, 13, 13, 13, 13, 13, 1

we do it for more values of $v$.

In [144]:
for v in 4:150
    if gcd(3,v)==1
    
    S=ulam(3,v,500)-list(3,v)
    println(S[1:50],",")
    end
    
end

[0, 0, 0, 0, -2, -3, -4, -6, -4, -6, -4, -4, -2, -4, -2, -2, 1, 1, 0, 3, 8, 6, 8, 8, 11, 11, 13, 11, 13, 13, 16, 18, 21, 21, 29, 29, 29, 29, 34, 32, 43, 42, 42, 45, 45, 53, 53, 53, 52, 50],
[0, 0, 0, 0, -1, -3, -3, -5, -6, -5, -6, -5, 1, -1, 1, -1, -1, 14, 15, 14, 12, 12, 24, 27, 25, 25, 34, 41, 48, 46, 54, 60, 59, 60, 58, 60, 67, 80, 89, 90, 92, 93, 93, 99, 103, 106, 106, 122, 122, 128],
[0, 0, 0, 0, 0, -2, -3, -3, -4, -6, -6, -7, -6, -7, -4, 2, 4, 5, 4, 4, 13, 21, 19, 18, 30, 30, 32, 30, 38, 38, 44, 44, 46, 44, 49, 55, 55, 69, 69, 69, 69, 69, 69, 72, 74, 77, 77, 92, 91, 91],
[0, 0, 0, 0, 0, -1, -3, -3, -3, -5, -6, -6, -8, -6, -8, -5, -2, -2, 1, 1, 2, 5, 4, 2, 5, 5, 5, 5, 14, 18, 27, 27, 27, 27, 25, 24, 25, 28, 31, 37, 37, 44, 44, 46, 44, 44, 44, 44, 44, 44],
[0, 0, 0, 0, 0, 0, -2, -3, -3, -3, -4, -6, -6, -7, -9, -7, -9, -7, -4, -1, -1, 2, 2, 7, 10, 10, 8, 10, 10, 10, 10, 10, 10, 19, 27, 36, 36, 36, 36, 36, 35, 33, 33, 32, 35, 38, 41, 47, 47, 58],
[0, 0, 0, 0, 0, 0, -1, -3, -3, -3, -3

We do the same with $u=7$

In [173]:
for v in 8:50
    
    if gcd(7,v)==1
        
        S=ulam(3,2*v+1,500)-list(7,v)
        println(S[1:40],",")
        
    end
end

[-4, 9, 5, 1, -3, -7, -11, -15, -20, -26, -30, -34, -38, -42, -46, -52, -57, -61, -67, -72, -76, -82, -87, -90, -95, -98, -99, -100, -101, -102, -100, -98, -102, -96, -97, -101, -100, -103, -68, -74],
[-4, 10, 6, 2, -2, -6, -10, -14, -18, -24, -29, -33, -37, -41, -45, -49, -54, -60, -64, -69, -75, -79, -84, -90, -94, -99, -102, -107, -108, -109, -110, -111, -112, -110, -108, -112, -102, -103, -107, -104],
[-4, 11, 7, 3, -1, -5, -9, -13, -17, -18, 11, 10, 6, 2, -2, -6, -7, 34, 30, 26, 22, 18, 14, 52, 66, 107, 103, 99, 95, 91, 129, 143, 181, 192, 230, 247, 282, 278, 286, 324],
[-4, 12, 8, 4, 0, -4, -8, -12, -16, -20, -25, -31, -35, -39, -43, -47, -51, -55, -59, -65, -70, -74, -80, -85, -89, -95, -100, -104, -110, -115, -118, -123, -126, -127, -128, -129, -130, -131, -132, -130],
[-4, 13, 9, 5, 1, -3, -7, -11, -15, -19, -23, -29, -34, -38, -42, -46, -50, -54, -58, -62, -67, -73, -77, -82, -88, -92, -97, -103, -107, -112, -118, -122, -127, -130, -135, -136, -137, -138, -139, -140],
[-4, 14

In [188]:
for v in 8:50
    if gcd(7,v)==1
        
        S=ulam(3,v,500)-list(7,v)
        println(S[1:40],",")
    end   
    
end

[-4, 0, -4, -8, -12, -17, -23, -27, -31, -37, -42, -46, -52, -54, -60, -61, -62, -66, -67, -71, -74, -75, -80, -86, -87, -91, -95, -99, -94, -94, -89, -93, -97, -101, -107, -112, -115, -116, -117, -115],
[-4, 0, -4, -8, -12, -13, -14, -15, -7, -2, -3, 5, 10, 9, 17, 22, 30, 35, 34, 42, 47, 55, 60, 59, 67, 72, 110, 106, 105, 125, 130, 144, 149, 163, 168, 170, 175, 174, 188, 208],
[-4, 0, -4, -8, -12, -16, -22, -27, -31, -35, -40, -46, -50, -55, -61, -63, -69, -71, -72, -73, -77, -78, -82, -81, -82, -86, -92, -94, -98, -102, -106, -110, -114, -109, -105, -100, -104, -108, -112, -116],
[-4, 0, -4, -8, -12, -16, -21, -27, -31, -35, -39, -45, -50, -54, -60, -65, -68, -73, -76, -77, -78, -76, -74, -78, -84, -85, -89, -94, -91, -74, -75, -81, -79, -83, -87, -85, -89, -93, -97, -91],
[-4, 0, -4, -8, -12, -16, -17, -6, -7, -11, -12, 11, 7, 3, 23, 28, 51, 47, 67, 72, 92, 94, 114, 122, 139, 135, 134, 154, 186, 182, 178, 201, 245, 259, 261, 287, 283, 282, 302, 334],
[-4, 0, -4, -8, -12, -16, -20, -

we do the same with $u=9$

In [187]:
for v in 8:50
    if gcd(9,v)==1
        
        S=ulam(9,v,500)-list(9,v)
        println(S[1:40],",")
        
    end
end

[0, 0, 0, -1, -9, -11, -18, -21, -27, -31, -38, -45, -50, -54, -60, -67, -74, -81, -88, -90, -98, -105, -112, -119, -116, -123, -130, -137, -134, -141, -148, -143, -150, -145, -152, -155, -146, -152, -152, -155],
[0, 0, 0, 0, -8, -9, -16, -18, -24, -27, -34, -41, -45, -49, -54, -61, -68, -75, -81, -83, -90, -97, -104, -111, -118, -126, -124, -131, -138, -145, -142, -149, -156, -163, -158, -165, -172, -165, -172, -163],
[0, 0, 0, 0, -7, -9, -14, -18, -21, -27, -32, -37, -45, -45, -53, -59, -64, -72, -78, -81, -86, -94, -100, -108, -114, -119, -126, -131, -136, -141, -146, -149, -154, -159, -164, -163, -168, -173, -168, -173],
[0, 0, 0, 0, -5, -9, -10, -18, -18, -24, -28, -36, -38, -45, -46, -54, -61, -65, -72, -75, -82, -90, -92, -99, -107, -111, -118, -126, -130, -137, -144, -147, -154, -157, -164, -165, -172, -175, -182, -183],
[0, 0, 0, 0, -4, -9, -9, -17, -18, -21, -27, -35, -36, -43, -45, -53, -56, -63, -70, -72, -80, -87, -90, -97, -100, -108, -116, -123, -126, -133, -140, -144, -

In [ ]:
#the first non-zero entry is in position 4,4,4,4,5,5,5,5,6,6,6,6,... for each sequence

## 3.4 Comparing Ulam(9,v) to the boring sequence 

In [13]:
for v in 10:50
    if gcd(9, v) == 1
        S = rangle_list(9, v, 500)[1:30]
        println(S,",")
    end
end

Any[0.5, 0.0, 0.5, 0.0, 0.5, 0.5, 0.5, 0.0, 0.5, 0.5, 0.5, 0.5, 0.0, 0.5, 0.5, 0.5, 0.5, 0.5, 0.0, 0.5, 0.5, 0.4999999999999929, 0.5, 0.5, 0.5000000000000071, 0.0, 0.5, 0.5, 0.5000000000000071, 0.5],
Any[0.47179275532556586, 0.021080034286803162, 0.49287278961236947, 0.9646655449379349, 0.5139528238991709, 0.4364583002635012, 0.5350328581859749, 0.9082510555890657, 0.5561128924727772, 0.380043810914632, 0.4786183688371075, 0.5771929267595794, 0.8518365662401983, 0.3236293215657646, 0.5982729610463835, 0.42220387948823657, 0.5207784374107121, 0.7954220768913309, 0.619352995333184, 0.26721483221689724, 0.3657893901393692, 0.640433029619988, 0.4643639480618411, 0.73900758754246, 0.5629385059843166, 0.6615130639067885, 0.21080034286802984, 0.3093749007905018, 0.4079494587129773, 0.5065240166354528],
Any[0.4756817676751641, 0.020429219975237167, 0.4961109876504022, 0.9717927553255645, 0.5165402076256385, 0.44747452300073043, 0.5369694276008765, 0.9231562906758946, 0.39883805835106045, 0.557

In [20]:
for v in 5:25#start at 5
    if gcd(9, 2*v+1) == 1 
        
        S = rangle_list(9, 2*v+1, 500)[1:30]
        println(S,",")
        
    end
end

Any[0.47179275532556586, 0.021080034286803162, 0.49287278961236947, 0.9646655449379349, 0.5139528238991709, 0.4364583002635012, 0.5350328581859749, 0.9082510555890657, 0.5561128924727772, 0.380043810914632, 0.4786183688371075, 0.5771929267595794, 0.8518365662401983, 0.3236293215657646, 0.5982729610463835, 0.42220387948823657, 0.5207784374107121, 0.7954220768913309, 0.619352995333184, 0.26721483221689724, 0.3657893901393692, 0.640433029619988, 0.4643639480618411, 0.73900758754246, 0.5629385059843166, 0.6615130639067885, 0.21080034286802984, 0.3093749007905018, 0.4079494587129773, 0.5065240166354528],
Any[0.4756817676751641, 0.020429219975237167, 0.4961109876504022, 0.9717927553255645, 0.5165402076256385, 0.44747452300073043, 0.5369694276008765, 0.9231562906758946, 0.39883805835106045, 0.5573986475761146, 0.48833296295120476, 0.8745198260262228, 0.5778278675513526, 0.35020159370138515, 0.43969649830153656, 0.8258833613765475, 0.5982570875265907, 0.5291914029016738, 0.30156512905171695, 0

In [21]:
for v in 5:50
    if gcd(9,1*v+1)==1
        S=ulam(9,2*v+1,500)-list(9,v)
        println(S[1:30],",")
    end
end

[0, 7, 7, 7, 2, -2, -3, -11, -11, -17, -21, -29, -31, -38, -39, -47, -54, -58, -65, -68, -75, -83, -85, -92, -100, -104, -111, -119, -123, -130],
[0, 8, 8, 8, 5, -1, -1, -7, -10, -7, -10, -7, 11, 5, 11, 5, 5, 50, 53, 50, 44, 44, 80, 89, 83, 83, 110, 131, 152, 146],
[0, 10, 10, 10, 10, 2, 1, 1, -6, -8, -8, -15, -23, -26, -26, -33, -40, -44, -44, -51, -58, -66, -71, -71, -78, -85, -92, -98, -98, -105],
[0, 11, 11, 11, 11, 5, 2, 2, -1, -7, -7, -10, -7, -10, -1, 17, 23, 26, 23, 23, 50, 74, 68, 65, 101, 101, 107, 101, 125, 125],
[0, 13, 13, 13, 13, 11, 4, 4, 4, 0, -5, -5, -9, -14, -20, -23, -27, -32, -32, -40, -45, -50, -50, -58, -63, -69, -77, -77, -85, -90],
[0, 14, 14, 14, 14, 23, 32, 41, 77, 104, 113, 149, 176, 185, 221, 248, 284, 311, 320, 356, 383, 419, 446, 455, 491, 518, 644, 644, 653, 725],
[0, 16, 16, 16, 16, 16, 11, 7, 7, 7, 6, -2, -2, -3, -11, -11, -17, -21, -29, -29, -30, -38, -40, -47, -48, -56, -58, -65, -66, -74],
[0, 17, 17, 17, 17, 17, 14, 8, 8, 8, 8, 2, -1, -1, -7, -10, -